src/plugins/home/server/services/sample_data/data_sets/util.test.ts

```diff
@@ -63,3 +63,100 @@ describe('getSavedObjectsWithDataSource()', () => {
+  ];
+  const generateTestDataSet = () => {
+    return {
+      id: 'foo',
+      name: 'Foo',
+      description: 'A test sample data set',
+      previewImagePath: '',
+      darkPreviewImagePath: '',
+      overviewDashboard: '',
+      getDataSourceIntegratedDashboard: () => '',
+      appLinks: [],
+      defaultIndex: '',
+      getDataSourceIntegratedDefaultIndex: () => '',
+      savedObjects,
+      getDataSourceIntegratedSavedObjects: (dataSourceId?: string, dataSourceTitle?: string) =>
+        savedObjects.map((item) => ({
+          ...item,
+          ...(dataSourceId ? { id: `${dataSourceId}_${item.id}` } : {}),
+          attributes: {
+            ...item.attributes,
+            title: dataSourceTitle
+              ? `${item.attributes.title}_${dataSourceTitle}`
+              : item.attributes.title,
+          },
+        })),
+      getWorkspaceIntegratedSavedObjects: (workspaceId?: string) =>
+        savedObjects.map((item) => ({
+          ...item,
+          ...(workspaceId ? { id: `${workspaceId}_${item.id}` } : {}),
+        })),
+      dataIndices: [],
+    };
+  };
+  it('should return consistent saved object id and title when workspace id and data source provided', () => {
+    expect(
+      getFinalSavedObjects({
+        dataset: generateTestDataSet(),
+        workspaceId: 'workspace-1',
+        dataSourceId: 'datasource-1',
+        dataSourceTitle: 'data source 1',
+      })
+    ).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          id: `workspace-1_datasource-1_saved-object-1`,
+          attributes: expect.objectContaining({
+            title: 'Saved object 1_data source 1',
+          }),
+        }),
+      ])
+    );
+  });
+  it('should return consistent saved object id when workspace id', () => {
+    expect(
+      getFinalSavedObjects({
+        dataset: generateTestDataSet(),
+        workspaceId: 'workspace-1',
+      })
+    ).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          id: `workspace-1_saved-object-1`,
+        }),
+      ])
+    );
+  });
+  it('should return consistent saved object id and title when data source id and title', () => {
+    expect(
+      getFinalSavedObjects({
+        dataset: generateTestDataSet(),
+        dataSourceId: 'data-source-1',
+        dataSourceTitle: 'data source 1',
+      })
+    ).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          id: `data-source-1_saved-object-1`,
+          attributes: expect.objectContaining({
+            title: 'Saved object 1_data source 1',
+          }),
+        }),
+      ])
+    );
+  });
+  it('should return original saved objects when no workspace and data source provided', () => {
+    const dataset = generateTestDataSet();
+    expect(
+      getFinalSavedObjects({
+        dataset,
+      })
+    ).toBe(dataset.savedObjects);
```

Comment:
should be "toEqual" here too, so can performs a deep equality check?

Reply:
Use toBe here means we just want to use the original saved objects here. I think toBe would be more strict than toEqual. toBe is for compare object reference, they are the same object if reference is same.